iina/DraggingDetect.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  DraggingDetect.swift
+//  iina
+//
+//  Created by Yuze Jiang on 05/08/2017.
+//  Copyright © 2017 lhc. All rights reserved.
+//
+
+import Foundation
+
+extension PlayerCore {
+
+  func openURLs(_ urls: [URL]) -> Int? {
+    let paths = urls.map{ $0.path }
+    if paths.count == 1 && ObjcUtils.isDirectory(paths[0]) {
+      if checkBDFolder(paths[0]) {
+        openURL(urls[0])
+        return nil
+      }
+    }
+    let (_, playableFiles) = checkPlayableFiles(paths, returnPaths: true)
+
+    let count = playableFiles.count
+    if count == 0 {
+      return 0
+    } else if count == 1 {
+      info.shouldAutoLoadFiles = true
+    } else {
+      info.shouldAutoLoadFiles = false
+    }
+    openURL(URL(fileURLWithPath: playableFiles[0], isDirectory: false))
+    for i in 1..<count {
+      addToPlaylist(playableFiles[i])
+    }
+    if count != 0 {
+      NotificationCenter.default.post(Notification(name: Constants.Noti.playlistChanged))
+    }
+    if count > 1 {
+      sendOSD(.addToPlaylist(count))
+    }
+    return count
+  }
+
+  func checkPlayableFiles(_ paths: [String], returnPaths: Bool = false) -> (Bool, [String]) {
+    var playableFiles: [String] = []
+    for path in paths {
+      if !ObjcUtils.isDirectory(path) {
+        if Utility.playableFileExt.contains((path as NSString).pathExtension.lowercased()) {
+          if returnPaths {
+            playableFiles.append(path)
+          } else {
+            return (true, [])
+          }
+        }
+      } else {
+        let dirEnumerator = FileManager.default.enumerator(atPath: path)
+        while let fileName = dirEnumerator?.nextObject() as? NSString {
+          if Utility.playableFileExt.contains(fileName.pathExtension.lowercased()) {
+            if returnPaths {
+              playableFiles.append(path + "/" + (fileName as String))
+            } else {
+              return (true, [])
+            }
+          }
+        }
+      }
+    }
+    return (!playableFiles.isEmpty, playableFiles)
```

Comment:
The first return value seems redundant since it can always be inferred from the second one